### PR TITLE
User/grahamc/220930 fix issue6761620

### PR DESCRIPTION
**Pull request by GrahamMcMynn**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

This change fixes this issue:
Task 6761620: Events not firing in child iFrames

### Main changes in the PR:

1. Events are proxied to child frames.

### Validation performed:

1. I ran a SharePoint app which IFrames a child frame and ensured that the back button/beforeUnload/apphover/appclick events were all proxied to the child Iframe.

### Unit Tests added:

Added a unit test that checked that an event was in both parent and child frames.

### End-to-end tests added:

No

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

Yes

### CODE REVIEW

File: packages/teams-js/src/internal/handlers.ts [hunk 1]
```diff
@@ -44,6 +44,9 @@ export function callHandler(name: string, args?: unknown[]): [true, unknown] | [
     callHandlerLogger('Invoking the registered handler for message %s with arguments %o', name, args);
     const result = handler.apply(this, args);
     return [true, result];
+  } else if (Communication.childWindow) {
+    sendMessageEventToChild(name, [args]);
+    return [true, undefined];
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/internal/handlers.ts:49]: I think this should be false instead of true, since the boolean signifies whether or not the handler was called and in this case it wasn't (it was proxied to the child window in case the child window has a handler for it)

File: packages/teams-js/src/public/pages.ts [hunk 2]
```diff
@@ -640,7 +642,12 @@ export namespace pages {
 
     function handleBackButtonPress(): void {
       if (!backButtonPressHandler || !backButtonPressHandler()) {
-        navigateBack();
+        if (Communication.childWindow) {
+          // If no parent window function handled it let the child window
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/pages.ts:646]: I think this should be current window, right? I would interpret "Parent" as "parent of the current window", which I don't think is what you mean

Comment (GrahamMcMynn) [reply packages/teams-js/src/public/pages.ts:646]: Yes, I will correct that comment.  I was thinking pare of the child window that I am sending it too.

File: packages/teams-js/src/internal/handlers.ts (hunk 1 above)
Comment (GrahamMcMynn) [reply packages/teams-js/src/internal/handlers.ts:49]: That seems reasonable.  I wasn't really sure what I should return, I will change it to false.

File: change/@microsoft-teams-js-7e2d7681-be7b-4c3c-b196-470aa65092c4.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Currently if a child IFrame is opened and the TeamsJs SDK is used it will not receive all the events that are sent to the parent IFrame.  This change allows the proxying of events if they are not handled in the parent frame.",
```
Comment (AE-MS) [inline change/@microsoft-teams-js-7e2d7681-be7b-4c3c-b196-470aa65092c4.json:3]: >Currently if a child IFrame is opened and the TeamsJs SDK is used it will not receive all the events that are sent to the parent IFrame.  This change allows the proxying of events if they are not handled in the parent frame. [](http://example.com/codeflow?start=14&length=224)

For consistency among changelog messages, can you rephrase to something like this (feel free to copy if you want):

"Enabled proxying of window events to child frames if unhandled by current frame."

**Review by AE-MS — CHANGES_REQUESTED** [review]
:clock1:

**Review by AE-MS — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by AE-MS — APPROVED** [review]
